components/wb-suggest/suggest.js

```diff
@@ -0,0 +1,279 @@
+/*
+ * Web Experience Toolkit (WET) / Boîte à outils de l'expérience Web (BOEW)
+ * wet-boew.github.io/wet-boew/License-en.html / wet-boew.github.io/wet-boew/Licence-fr.html
+ *
+ * @author: duboisp
+ * @description: Load suggestions from an array of string in a JSON file.
+ *
+ */
+( function( $, document, wb ) {
+"use strict";
+
+var $document = wb.doc,
+	componentName = "wb-suggest",
+	selector = "[data-" + componentName + "]",
+	initEvent = "wb-init." + componentName,
+	jsonFetched = "json-fetched.wb",
+	wait,
+	waitInterval = 250, // In-bettween typing delay before refreshing the suggested list.
+	maxWaitLoading = 5, // Number of time of waitInterval the plugin are allow wait for getting JSON suggestions
+
+	// Remove accent and normalize the string
+	//
+	// str: String to be normalized
+	// return: A normalized string with no accent
+	//
+	unAccent = function( str ) {
+		return str.normalize( "NFD" ).replace( /[\u0300-\u036f]/g, "" );
+	},
+
+	// Remove children of an element except template element
+	//
+	// this: element that need to emptied
+	//
+	emptyExceptTemplate = function() {
+		var elm,
+			children = this.children,
+			i;
+
+		for ( i = children.length - 1; i > 0; i = i - 1 ) {
+			elm = children[ i ];
+			if ( elm.nodeType === 1 && elm.nodeName !== "TEMPLATE" ) {
+				this.removeChild( elm );
+			}
+
+		}
+	},
+
+	// Add suggested options to the datalist
+	//
+	// this: datalist instance
+	// filter: filter items that match the suggestion
+	// limti: (for overwride) limit number of result
+	// attSuggestions: (for overwride) Array of string with suggestion
+	//
+	addDataListOptions = function( filter, limit, attrSuggestions ) {
+		var suggestions = attrSuggestions || JSON.parse( this.dataset.wbSuggestions || [] ),
+			filterType = this.dataset.wbFilterType || "any",
+			filterRegExp,
+			lenSuggestions = suggestions.length,
+			idx, suggestion, suggestionNorm,
+			clone, option,
+			optionAddedUnAccent = [],
+			currentOptions = this.childNodes,
+			j, j_cache, j_len = currentOptions.length - 1,
```

Comment:
I know it would probably be in extreme cases, but could the currentOptions length be 0 and give a result of -1, potentially causing an infinite loop at line 106? If it is a concern, maybe just add a simple check for length greater than 0 or else leave as is.

Reply:
This didn't happen during testing